src/main/java/org/apache/maven/resolver/internal/ant/types/model/Repository.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.resolver.internal.ant.types.model;
+
+import org.apache.tools.ant.types.DataType;
+
+/**
+ * Represents a repository in a project model.
+ */
+public class Repository extends DataType {
+    private Id id;
+    private Name name;
+    private Url url;
```

Comment:
I do not like using a Url that is not a java.net.URL. This is confusing. Can we just use a String? or rename this class to something like UrlElement?

Reply:
The reason is to keep the same names as in the POM file. Making an exception just for Url so it would be named something else would be very confusing I think.

Reply:
Please don't resolve comments. That just means I have to open them to see what you said.